System/Console/ANSI/Windows/Emulator.hs

```diff
@@ -312,9 +309,8 @@ applyANSISGRToAttribute def sgr attribute = case sgr of
   iNTENSITY = fOREGROUND_INTENSITY
 
 approximateColor8Code :: Color8Code -> (ColorIntensity, Color)
-approximateColor8Code code@(Color8Code c)
-  | c < 0 || c > 255 = (Vivid, Red) -- safe fallback to avoid interrupting the program with error
-  | otherwise        = either (id) (closest4bitsANSIColor . color8To24) $ color8CodeToEither code
+approximateColor8Code code
+  = closest4bitsANSIColor $ xterm256ToSRGB $ color8CodeToXterm256 code
```

Comment:
It would be nice to test that for grayscale colors, it returns white, black or one of the 2 grays in the 4bits color. I suspect that it could return a color instead for some values.

Reply:
Agreed. I have access to a PC running legacy Windows during the week - I'll try to test it then.